tokenizers/src/models/bpe/trainer.rs

```diff
@@ -8,6 +8,63 @@ use serde::{Deserialize, Serialize};
 use std::cmp::Ordering;
 use std::collections::{BinaryHeap, HashMap, HashSet};
 
+mod unsafevec {
+    use std::cell::UnsafeCell;
+    use std::marker::PhantomData;
+
+    use super::Word;
+
+    pub(super) struct UnsafeVec<'v> {
+        value: UnsafeCell<*mut Word>,
```

Comment:
I don't think this UnsafeCell is doing anything here?

Reply:
I wasn't certain when I wrote this, does the borrow on self of `fn get_unchecked_mut` not affect anything?
One is not normally permitted to mutate past a &, you can't use a `&&mut T` to mutate `T`, so I suspected one cannot easily mutate `T` behind `&*mut T`. Miri seems fine with it, so I guess this UnsafeCell is not needed and you're correct.